platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/ExtensibleDynamicObject.java

```diff
@@ -134,11 +135,10 @@ public void setParent(DynamicObject parent) {
     @Override
     @Deprecated
     public org.gradle.api.plugins.Convention getConvention() {
-// TODO nag once KGP doesn't register conventions anymore
-//        DeprecationLogger.deprecateType(org.gradle.api.internal.HasConvention.class)
-//            .willBeRemovedInGradle9()
-//            .withUpgradeGuideSection(8, "deprecated_access_to_conventions")
-//            .nagUser();
+        DeprecationLogger.deprecateType(org.gradle.api.internal.HasConvention.class)
```

Comment:
❓ Would the deprecation of the internal type be clear to users?
For instance, they'd see this deprecation if their build logic calls out to the `Project.getConvention()`, but that method is no way connected to the `HasConvention` publicly.

Would it be simpler to use only the deprecation for `...api.plugins.Convention` ?

Reply:
Good idea., and it has the benefit of deduplicating nested deprecations.
Fixed, PTAL